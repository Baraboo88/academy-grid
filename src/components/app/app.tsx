import { ThemeProvider } from 'styled-components';
import {
  Switch,
  Route,
  BrowserRouter as Router,
} from '../common/common';

import DetailedQuest from '../detailed-quest/detailed-quest';
import Contacts from '../contacts/contacts';
import Home from '../home/home';
import { appTheme } from './common';
import * as S from './app.styled';
import React from 'react';
import NotFound from '../not-found/not-found';

const App = () => (
  <ThemeProvider theme={appTheme}>
    <S.GlobalStyle />
    <Router>
      <Switch>
        <Route exact path="/detailed-quest/:id" component={DetailedQuest}/>
        <Route exact path="/contacts">
          <Contacts />
        </Route>
        <Route exact path="/" >
          <Home />
        </Route>
        <Route>
          <NotFound/>
        </Route>
      </Switch>
    </Router>
  </ThemeProvider>
);

export default App;
