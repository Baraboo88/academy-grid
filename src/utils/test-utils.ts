import { QuestLevel, QuestModel, QuestType } from './utils';
import { createApi } from '../api';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';


export const mockQuests: QuestModel[]  = [
    {
      id: 1,
      "title": "Склеп",
      "description": "Средневековое кладбище таит в себе много страшных тайн. Местные жители говорят, что в склепе похоронен граф вампир, который по ночам выходит на охоту, чтобы испить человеческой крови. Через час солнце опустится за горизонт, успеете ли вы убить вампира и выбраться из склепа?",
      "previewImg": "img/preview-sklep.jpg",
      "coverImg": "img/cover-sklep.jpg",
      type: "horror" as QuestType,
      level: "hard" as QuestLevel,
      peopleCount: [2, 5],
      duration: 120
    },
    {
      "id": 2,
      "title": "Маньяк",
      "description": "В комнате с приглушённым светом несколько человек, незнакомых друг с другом, приходят в себя. Никто не помнит, что произошло прошлым вечером. Руки и ноги связаны, но одному из вас получилось освободиться. На стене висит пугающий таймер и запущен отсчёт 60 минут. Сможете ли вы разобраться в стрессовой ситуации, помочь другим, разобраться что произошло и выбраться из комнаты?",
      "previewImg": "img/preview-maniac.jpg",
      "coverImg": "img/cover-maniac.jpg",
      "type": "horror" as QuestType,
      "level": "medium" as QuestLevel,
      "peopleCount": [3, 6],
      "duration": 90
    },
    {
      "id": 3,
      "title": "Ритуал",
      "description": "Тяжелый воздух угнетает, в ночи вы оказываетесь запертыми в сыром помещении вместе с другими ничего не понимающими жертвами. Сквозь щель в двери вы видите, как некто в капюшоне готовит площадку как будто для проведения мистического обряда. Удастся ли вам выбраться, пока вы не станете жертвой ритуала?",
      "previewImg": "img/preview-ritual.jpg",
      "coverImg": "img/cover-ritual.jpg",
      "type": "mystic" as QuestType,
      "level": "hard" as QuestLevel,
      "peopleCount": [3, 5],
      "duration": 120
    },
    {
      "id": 4,
      "title": "Тайны старого особняка",
      "description": "Погрузитесь в атмосферу служебных помещений закулисья, которые хранят множество тайн и загадок. Вы окажитесь в старом особняке и увидите все, что скрывают его запутанные коридоры.",
      "previewImg": "img/preview-final-frontier.jpg",
      "coverImg": "img/cover-final-frontier.jpg",
      "type": "detective" as QuestType,
      "level": "easy" as QuestLevel,
      "peopleCount": [2, 5],
      "duration": 60
    },
    {
      "id": 5,
      "title": "Хижина в лесу",
      "description": "Вы с друзьями оказались в заброшенной хижине. Какую тайну она скрывает и как из неё выбраться? На эти вопросы вам предстоит найти ответ, чтобы вернуться домой.",
      "previewImg": "img/preview-house-in-the-woods.jpg",
      "coverImg": "img/cover-house-in-the-woods.jpg",
      "type": "mystic" as QuestType,
      "level": "medium" as QuestLevel,
      "peopleCount": [4, 7],
      "duration": 90
    },
    {
      "id": 6,
      "title": "Фатальный эксперимент",
      "description": "Вы стоите на пороге нового научного открытия, которое перевернет судьбу человечества. Но что-то идёт не так, и ядерный реактор, который работает на полную мощность, сигнализирует о скорой поломке. Удастся ли вам починить его в отведенное время и предотвратить гибель людей в этом фатальном эксперименте?",
      "previewImg": "img/preview-fatal-exp.jpg",
      "coverImg": "img/cover-fatal-exp.jpg",
      "type": "adventures" as QuestType,
      "level": "hard" as QuestLevel,
      "peopleCount": [5, 8],
      "duration": 120
    },

  ]
;


export const getTestStore = () => {
  const initialState = {
    quests: mockQuests,
    currentQuest: mockQuests[1],
    errorMsg: ''
  };
  const reducer = (state = initialState) => {
    return state;
  };
  const api = createApi();

  return createStore(reducer, applyMiddleware(thunk.withExtraArgument(api)));
};
