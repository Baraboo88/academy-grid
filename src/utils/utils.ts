import { ActiveTab } from '../reducer/reducer';

export enum QuestType {
  Adventures = 'adventures',
  Horror = 'horror',
  Mystic = 'mystic',
  Detective = 'detective',
  SciFi='sci-fi'
}

export enum QuestLevel {
  Easy = 'easy',
  Medium = 'medium',
  Hard = 'hard'
}

export enum QuestDuration {
  Sixty = 60,
  Ninety = 90,
  OneTwoZero = 120
}

export interface QuestModel {
  id: number;
  title: string;
  description: string;
  previewImg: string;
  coverImg: string;
  type: QuestType;
  level: QuestLevel;
  peopleCount: number [];
  duration: QuestDuration;
}

export interface OrderModel {
  name: string;
  peopleCount: number;
  phone: string;
  isLegal: boolean
}

export interface StateModel {
  quests: QuestModel [];
  currentQuest?: QuestModel;
  isOrderSent?: boolean;
  isResponseReceived: boolean;
  errorMsg: string;
  activeTab: ActiveTab;
}


export const getCyrillicLevel = (level: QuestLevel) => {
  switch (level) {
    case QuestLevel.Easy:
      return 'простой';
    case QuestLevel.Medium:
      return 'средний';
    case QuestLevel.Hard:
      return 'сложный';
  }
  return '';
};

export const getCyrillicType = (questType: QuestType) => {
  switch (questType) {
    case QuestType.Adventures:
      return 'Приключения';
    case QuestType.Horror:
      return 'Ужасы';
    case QuestType.Mystic:
      return 'Мистика';
    case QuestType.Detective:
      return 'Детектив';
    case QuestType.SciFi:
      return 'Sci-fi';
  }
  return '';
};


export const applyFilter = (quests: QuestModel[], filter: QuestType) => quests.filter((quest) => quest.type === filter);
