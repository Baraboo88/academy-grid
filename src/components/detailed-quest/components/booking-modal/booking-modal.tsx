import * as S from './booking-modal.styled';
import { ReactComponent as IconClose } from 'assets/img/icon-close.svg';
import React, { useEffect, useState } from 'react';
import { ActionCreator, Operation } from '../../../../reducer/reducer';
import { connect } from 'react-redux';
import { OrderModel, StateModel } from '../../../../utils/utils';
import { getErrorMsg, getIsOrderSent } from '../../../../reducer/selectors';
import { AlertMsg } from '../../../home/home.styled';

interface BookingModalProps {
  onBookingModalClose: () => void;
  isOrderSent: boolean;
  errorMsg: string;
  sendOrder: (order: OrderModel) => void;
  setIsOrderSent: (isOrderSent: boolean) => void;
}

const BookingModal: React.FC<BookingModalProps> = (props) => {
  const { onBookingModalClose, isOrderSent, errorMsg, sendOrder,setIsOrderSent  } = props;

  useEffect(() => {
    if (isOrderSent) {
      setName('');
      setPeopleCount('');
      setPhone('');
      setIsLegal(false);
      onBookingModalClose();
    }
    if(errorMsg) {
      setError(errorMsg);
    }
    return () => {
      setIsOrderSent(false);
    }
  }, [isOrderSent, errorMsg, onBookingModalClose, setIsOrderSent]);

  const [error, setError] = useState('');
  const [name, setName] = useState<string>('');
  const [peopleCount, setPeopleCount] = useState<string>('');
  const [phone, setPhone] = useState<string>('');
  const [isLegal, setIsLegal] = useState<boolean>(false);

  const onFormSubmit = () => {
    sendOrder({ peopleCount: Number(peopleCount), isLegal, phone, name });
  };

  const onChangeLegal = () => {
    setIsLegal(!isLegal);
  };

  return (
    <S.BlockLayer>
      <S.Modal>
        <S.ModalCloseBtn onClick={onBookingModalClose} data-test='test-model-close'>
          <IconClose width='16' height='16' />
          <S.ModalCloseLabel>Закрыть окно</S.ModalCloseLabel>
        </S.ModalCloseBtn>
        <S.ModalTitle>Оставить заявку</S.ModalTitle>
        <S.BookingForm
          action='https://echo.htmlacademy.ru'
          method='post'
          id='booking-form'
          onSubmit={(evt) => {
            evt.preventDefault();
            onFormSubmit();
            }
          }
          data-test="test-addOrder"
        >
          <S.BookingField>
            <S.BookingLabel htmlFor='booking-name'>Ваше Имя</S.BookingLabel>
            <S.BookingInput
              type='text'
              id='booking-name'
              name='booking-name'
              placeholder='Имя'
              required
              value={name}
              onChange={(evt) => {
                setName(evt.target.value);
                setError('');
              }
              }
            />
          </S.BookingField>
          <S.BookingField>
            <S.BookingLabel htmlFor='booking-phone'>
              Контактный телефон
            </S.BookingLabel>
            <S.BookingInput
              type='tel'
              id='booking-phone'
              name='booking-phone'
              placeholder='Телефон'
              required
              value={phone}
              onChange={(evt) => {
                setPhone(evt.target.value);
                setError('');
              }
              }
            />
          </S.BookingField>
          <S.BookingField>
            <S.BookingLabel htmlFor='booking-people'>
              Количество участников
            </S.BookingLabel>
            <S.BookingInput
              type='number'
              id='booking-people'
              name='booking-people'
              placeholder='Количество участников'
              required
              value={peopleCount}
              onChange={(evt) => {
                setPeopleCount(evt.target.value);
                setError('');
              }}
            />
          </S.BookingField>
          {error && (
            <AlertMsg>
              {error}
            </AlertMsg>
          )}
          <S.BookingSubmit type='submit'>Отправить заявку</S.BookingSubmit>
          <S.BookingCheckboxWrapper>
            <S.BookingCheckboxInput
              type='checkbox'
              id='booking-legal'
              name='booking-legal'
              required
              checked={isLegal}
              onChange={onChangeLegal}
            />

            <S.BookingCheckboxLabel
              className='checkbox-label'
              htmlFor='booking-legal'
            >
              <S.BookingCheckboxText>
                Я согласен с{' '}
                <S.BookingLegalLink href='#'>
                  правилами обработки персональных данных и пользовательским
                  соглашением
                </S.BookingLegalLink>
              </S.BookingCheckboxText>
            </S.BookingCheckboxLabel>
          </S.BookingCheckboxWrapper>
        </S.BookingForm>
      </S.Modal>
    </S.BlockLayer>
  );
};


const mapStateToProps = (state: StateModel) => {
  return {
    isOrderSent: getIsOrderSent(state),
    errorMsg: getErrorMsg(state),
  };
};


const mapDispatchToProps = (dispatch: any) => ({
  sendOrder(order: OrderModel) {
    dispatch(Operation.sendOrder(order));
  },
  setIsOrderSent(isOrderSent: boolean) {
    dispatch(ActionCreator.setIsOrderSent(isOrderSent))
  }
});


export default connect(mapStateToProps, mapDispatchToProps)(BookingModal);
export { BookingModal };
