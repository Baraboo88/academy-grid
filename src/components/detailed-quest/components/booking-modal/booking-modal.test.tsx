import * as Enzyme from 'enzyme';
import { mount } from 'enzyme';
import EnzymeReactAdapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';
import { getTestStore } from '../../../../utils/test-utils';
import { appTheme } from '../../../app/common';

import { BrowserRouter } from 'react-router-dom';
import { BookingModal } from './booking-modal';


Enzyme.configure({ adapter: new EnzymeReactAdapter() });




it(`BookingModal successfully rendered`, () => {
  const MOCK_IS_ORDER_SENT = true
   const MOCK_BOOKING_MODAL_ERROR = ''
  const mockModalCloseHandler = jest.fn();
  const mockSendOrderHandler = jest.fn();
  const mockSetIsOrderSent = jest.fn();
  const tree = mount(<Provider store={getTestStore()}><BrowserRouter><ThemeProvider
    theme={appTheme}><BookingModal setIsOrderSent={mockSetIsOrderSent} errorMsg={MOCK_BOOKING_MODAL_ERROR}
                                   onBookingModalClose={mockModalCloseHandler} isOrderSent={MOCK_IS_ORDER_SENT}
                                   sendOrder={mockSendOrderHandler} /></ThemeProvider></BrowserRouter></Provider>);
  expect(toJson(tree, { mode: `deep` })).toMatchSnapshot();
});
