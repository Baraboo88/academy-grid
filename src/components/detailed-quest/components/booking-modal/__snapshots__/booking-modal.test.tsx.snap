// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingModal successfully rendered 1`] = `
.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 100%;
  margin: 0;
  padding-top: 22px;
  padding-right: 47px;
  padding-bottom: 23px;
  padding-left: 48px;
  font-family: inherit;
  font-size: 17px;
  line-height: 20px;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  font-weight: 800;
  text-transform: uppercase;
  color: #FFFFFF;
  background-color: #F2890F;
  border: none;
  border-radius: 65px;
  cursor: pointer;
}

.c8:focus,
.c8:hover {
  background-color: #F39425;
}

.c8:active {
  opacity: 0.8;
}

.c0 {
  z-index: 10;
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background-color: rgba(61,51,51,0.96);
}

.c1 {
  position: absolute;
  top: 50%;
  left: 50%;
  max-height: 100%;
  width: 480px;
  padding: 32px;
  padding-bottom: 50px;
  overflow-y: auto;
  overflow-x: hidden;
  background-color: #1F1D1D;
  -webkit-transform: translateX(-50%) translateY(-50%);
  -ms-transform: translateX(-50%) translateY(-50%);
  transform: translateX(-50%) translateY(-50%);
  border-radius: 3px;
}

.c2 {
  position: absolute;
  top: 29px;
  right: 22px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px;
  background: none;
  border: none;
  font: inherit;
  cursor: pointer;
}

.c2:focus svg,
.c2:hover svg {
  opacity: 0.5;
}

.c2:active svg {
  opacity: 0.3;
}

.c3 {
  margin: 0;
  margin-bottom: 39px;
  padding: 0;
  font-size: 32px;
  line-height: 120%;
  font-weight: 800;
  color: #FFFFFF;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 0;
  padding-right: 16px;
}

.c5:not(:last-of-type) {
  margin-bottom: 30px;
}

.c6 {
  margin-bottom: 15px;
  color: #E6E6E6;
}

.c7 {
  padding-top: 17px;
  padding-right: 24px;
  padding-bottom: 18px;
  padding-left: 23px;
  font-family: inherit;
  color: #FFFFFF;
  background-color: transparent;
  border: 1px solid rgba(255,255,255,0.4);
  border-radius: 3px;
}

.c7::-webkit-input-placeholder {
  color: #E5E5E5;
  opacity: 0.64;
}

.c7::-moz-placeholder {
  color: #E5E5E5;
  opacity: 0.64;
}

.c7:-ms-input-placeholder {
  color: #E5E5E5;
  opacity: 0.64;
}

.c7::placeholder {
  color: #E5E5E5;
  opacity: 0.64;
}

.c7:focus,
.c7:hover {
  opacity: 0.8;
}

.c9 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin-top: 55px;
  margin-bottom: 21px;
  padding-top: 15px;
  padding-right: 34px;
  padding-bottom: 16px;
  padding-left: 34px;
  font-size: 14px;
  line-height: 16px;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  background-color: #B8B8B8;
}

.c9:focus,
.c9:hover {
  background-color: rgba(255,255,255,0.32);
}

.c10 {
  position: relative;
  margin: 0;
  padding: 0;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  padding-left: 26px;
  cursor: pointer;
}

.c12::before,
.c12::after {
  position: absolute;
  top: 3px;
  left: 2px;
  width: 16px;
  height: 16px;
  opacity: 0.4;
}

.c12::before {
  content: '';
  background-color: #F2890F;
  border-radius: 4px;
}

.c12:hover {
  opacity: 0.8;
}

.c12:active {
  opacity: 0.6;
}

.c11 {
  top: 20px;
  left: 10px;
}

.c11:checked ~ .checkbox-label::after {
  content: '';
  background-image: url(icon-tick.svg);
  background-repeat: no-repeat;
  background-position: center;
}

.c11:focus ~ .checkbox-label {
  outline: 2px solid #4d90fe;
  outline-offset: 3px;
  opacity: 0.8;
}

.c11:active ~ .checkbox-label {
  opacity: 0.6;
}

.c13 {
  font-size: 14px;
  line-height: 144%;
  color: #E6E6E6;
}

.c14 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div
  className="c0"
>
  <section
    className="c1"
  >
    <button
      className="c2"
      data-test="test-model-close"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      <svg
        height="16"
        width="16"
      >
        icon-close.svg
      </svg>
      <span
        className="visually-hidden"
      >
        Закрыть окно
      </span>
    </button>
    <h2
      className="c3"
    >
      Оставить заявку
    </h2>
    <form
      action="https://echo.htmlacademy.ru"
      className="c4"
      data-test="test-addOrder"
      id="booking-form"
      method="post"
      onSubmit={[Function]}
    >
      <p
        className="c5"
      >
        <label
          className="c6"
          htmlFor="booking-name"
        >
          Ваше Имя
        </label>
        <input
          className="c7"
          id="booking-name"
          name="booking-name"
          onChange={[Function]}
          placeholder="Имя"
          required={true}
          type="text"
          value=""
        />
      </p>
      <p
        className="c5"
      >
        <label
          className="c6"
          htmlFor="booking-phone"
        >
          Контактный телефон
        </label>
        <input
          className="c7"
          id="booking-phone"
          name="booking-phone"
          onChange={[Function]}
          placeholder="Телефон"
          required={true}
          type="tel"
          value=""
        />
      </p>
      <p
        className="c5"
      >
        <label
          className="c6"
          htmlFor="booking-people"
        >
          Количество участников
        </label>
        <input
          className="c7"
          id="booking-people"
          name="booking-people"
          onChange={[Function]}
          placeholder="Количество участников"
          required={true}
          type="number"
          value=""
        />
      </p>
      <button
        className="c8 c9"
        type="submit"
      >
        Отправить заявку
      </button>
      <div
        className="c10"
      >
        <input
          checked={false}
          className="c11 visually-hidden"
          id="booking-legal"
          name="booking-legal"
          onChange={[Function]}
          required={true}
          type="checkbox"
        />
        <label
          className="c12 checkbox-label"
          htmlFor="booking-legal"
        >
          <span
            className="c13"
          >
            Я согласен с
             
            <a
              className="c14"
              href="#"
            >
              правилами обработки персональных данных и пользовательским соглашением
            </a>
          </span>
        </label>
      </div>
    </form>
  </section>
</div>
`;
